doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+
+## 2.1 ConfigDB Changes
+
+**SRV6_SID_LIST_TABLE**
+
+Description: New table that stores SRv6 SID list configuration.
+
+Schema:
+
+```
+; New table
+; holds SRv6 SID list
+
+key = SRV6_SID_LIST:segment_name
+                     ; SID segment name
+; field = value
+path = SID,          ; List of SIDs
+
+For example:
+    "SRV6_SID_LIST": {
+        "seg1": {
+            "path": [
+                "BABA:1001:0:10::",
+                "BABA:1001:0:20:F1::"
+            ]
+        },
+        "seg2": {
+            "path": [
+                "BABA:1001:0:30::",
+                "BABA:1001:0:40:F1::"
+            ]
+        }
+    }
+```
+
+**SRV6_LOCAL_SID_TABLE**
+
+Description: New table to hold local SID to behavior mapping
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping, allow 1:1 or n:1 mapping
+
+key = SRV6_LOCAL_SID:ipv6address
+; field = value
+block_len = blen             ; bit length of block portion in address, default 40
+node_len = nlen              ; bit length of node ID portion in address, default 24
+func_len = flen              ; bit length of function portion in address, default 16
+arg_len = alen               ; bit length of argument portion in address
+action = behavior            ; behaviors defined for local SID
+vrf = VRF_TABLE.key          ; VRF name for END.DT46, can be empty
+adj = address,               ; Optional, list of adjacencies for END.X
+policy = SRV6_POLICY.key     ; Optional, policy name for END.B6.ENCAP
+source  = address,           ; Optional, list of src addrs for encap for END.B6.ENCAP
+weight  = weight.value,      ; Optional, list of assigned weight
+
+For example:
+    "SRV6_LOCAL_SID" : {
+       	"BABA:1001:0:20:F1::" : {
+           "action": "end.dt46",
+           "vrf":  "VRF-1001"
+        },
+       	"BABA:1001:0:40:F1::" : {
+           "action": "end.dt46",
+           "vrf":  "VRF-1001"
+        },
+       	"BABA:1001:0:20:F2::" : {
+           "action": "end.x",
+           "adj": [
+                BABA:2001:0:10::1, 
+                BABA:2001:0:10::2
+            ],
+            "weight": "1,2"
+        },
+       	"BABA:1001:0:20:F3::" : {
+           "action": "end.b6.encap",
+           "segment": "seg1"
+           "source": "A::1"
+        }
+    }
+```
+
+**SRV6_POLICY_TABLE**
+
+Description: New table that stores SRv6 policy .
+
+Schema:
+
+```
+; New table
+; holds SRv6 policy
+
+key = SRV6_POLICY:policy_name
+
+; field = value
+endpoint = address             ; 
+segment = SRv6_SID_LIST.key,   ; List of segment names
+weight  = weight.value,        ; List of assigned weight value
+
+For example:
+    "SRV6_POLICY": {
+        "policy1": {
+            "segment": "seg1, seg2",
+            "weight": "1, 2"
+            },
+        "policy2": {
+            “endpoint": "BABA:1001:0:40::1"
+            "segment": "seg1"
+        }
+    }
+```
+
+**SRV6_STEER_MAP**
+
+Description: New table that stores prefix to policy mapping .
+
+Schema:
+
+```
```

Comment:
Please create SONiC YANG PR for review if not done already.

Reply:
provided in the latest HLD

Reply:
@venkatmahalingam @prsunny Could you please take a look to review the SRv6 HLD. Thanks.